### PR TITLE
feat: add problem interaction charts to superset (FC-0024)

### PR DESCRIPTION
**Pull request by SoryRawyer**
Also breaks up the instructor dashboard into separate tabs for enrollment, video, and problem charts.

I realize this PR would have a number of conflicts should the assets refactor land before this, but I wanted to open this PR to start getting feedback. I'll resolve the merge conflicts once that PR lands.

Below are screenshots of the problem interaction tab, specifically.

image

image

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @SoryRawyer!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: tutoraspects/templates/aspects/apps/superset/pythonpath/queries/dim_course_videos.sql
Comment (SoryRawyer) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/queries/dim_course_videos.sql:30]: I don't have a clean reproduction of the issue I was seeing, but I ran into cases where joins that used the `using` syntax were failing with "column not found" errors. I'll see if I can put together a ClickHouse fiddle with the behavior but in the meantime I wanted to get these changes in.

Comment (Ian2012) [reply tutoraspects/templates/aspects/apps/superset/pythonpath/queries/dim_course_videos.sql:30]: With the instructor dashboard, I've experienced weird cases with the demo data. Some behaviors:

- Dashboard is blank, with just filters and no charts, just a loading screen.
- Some charts randomly fail. Needs to force refresh to appear
- As the org filter is required, and the filters are cached, no organization appears in the selector. Not sure if it's a bug with the demo data or superset cache.

**Comment by Ian2012** [comment]
As for the dashboard, I really really like the new disposal. Great work, waiting for a code review to approve it

File: tutoraspects/templates/aspects/apps/superset/pythonpath/queries/dim_course_videos.sql
Comment (SoryRawyer) [reply tutoraspects/templates/aspects/apps/superset/pythonpath/queries/dim_course_videos.sql:30]: Does it seem like this is happening only when you merge new changes, or does it happen even if you just look at the dashboard without changing anything?

Since I'm also working with the demo course data, my guess is that it's related to the Superset cache. I've noticed some odd behavior after importing assets. Hopefully this is just related to refreshing the assets and occurs less frequently once the changes slow down.

**Review by bmtcril — APPROVED** [review]
I can't test loading this right now, but it all looks reasonable and the tabs are great. I'm also seeing some of the same things you both were in regards to glitchy loading. It seemed to settle on a restart of Superset, so that does point to a caching thing.

Let's get it in and tinker with it for a bit to get a feel for where we'd like to take it next!

**Comment by SoryRawyer** [comment]
Alright, I've been able to successfully import assets after merging in the assets refactor.

**Comment by bmtcril** [comment]
Nice, I'll merge it once our beefy new tests finish up

**Comment by openedx-webhooks** [comment]
@SoryRawyer 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/154&entry.1484655318=https://github.com/SoryRawyer&entry.752974735=2023-07-06+23:01&entry.1917517419=2023-07-07+17:18&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
